cmd/karmada-search/app/options/options.go

```diff
@@ -83,123 +61,3 @@ func (o *Options) AddFlags(flags *pflag.FlagSet) {
 func (o *Options) Complete() error {
 	return nil
 }
-
-// Run runs the aggregated-apiserver with options. This should never exit.
-func (o *Options) Run(ctx context.Context) error {
```

Comment:
# The Cyclic Import Problem

(1) Option is defined in `github.com/karmada-io/karmada/cmd/karmada-search/app`

```
// Option configures a framework.Registry.
type Option func(*runtime.Registry)
```

(2) `github.com/karmada-io/karmada/cmd/karmada-search/app` imports `"github.com/karmada-io/karmada/cmd/karmada-search/app/options"`

(3) I need to pass `type Option` to `Run()`, this will make `"github.com/karmada-io/karmada/cmd/karmada-search/app/options"` import `github.com/karmada-io/karmada/cmd/karmada-search/app`.

This will introduce cyclic import between packages, go compiler doesn't allow this.

# Style Reference

This is also official Kubernetes style. See

https://github.com/kubernetes/kubernetes/blob/4c6338ac0f66001ae0afa14f1b7a3b285accd97a/cmd/kube-scheduler/app/server.go#L118

Reply:
ok, thanks for explain.